modules/system/checks.nix

```diff
@@ -258,6 +259,12 @@ in
       description = "Whether to run the Nix build users validation checks.";
     };
 
+    system.checks.verifyBuildGroup = mkOption {
+      type = types.bool;
+      default = config.nix.enable;
+      description = "Whether to run the Nix build group validation checks.";
+    };
+
```

Comment:
I think we should probably just make this unconditional on `config.nix.enable`, since the group creation is. It’s not ideal if the GID mismatches since automatically changing the `nixbld` GID is a pain.